src/record/record.js

```diff
@@ -449,7 +446,16 @@ Record.prototype._applyUpdate = function( message ) {
  */
 Record.prototype._onRead = function( message ) {
 	this.version = parseInt( message.data[ 1 ], 10 );
-	this._applyChange( jsonPath.set( this._$data, undefined, JSON.parse( message.data[ 2 ] ) ) );
+
+	this._$data = JSON.parse( message.data[ 2 ] );
```

Comment:
@ronag is the change from `_applyChange` to doing it all inline an optimisation? Using applyChange would stop it from triggering undefined on the all the paths that don't exist yet on the first update.